### ISSUE TITLE
kui like CLI and endpoint?

### ISSUE THREAD

**Issue by jstrachan** [vmware-tanzu/octant#2220]
this looks an interesting approach:
https://github.com/kubernetes-sigs/kui

making it easy to open a UI from kubectl to visualise results and essentially being able to run `kubectl get .....` and visualise the results in the UI.

I wonder if it might make sense to add something like this to octant? so we could do something like `octant get pod -n myns -l foo=bar` and it was like `kubectl get .....` only it opened a new window in octant to visualise the results? 

Another aspect of kui thats interesting is being able to have multiple windows running `kubectl get` commands and then letting folks visualise/navigate the data. Maybe something like that could be added too to help CLI folks navigate quickly using k8s style CLIs?

**Comment by jstrachan** [comment]
given the k8s CLI muscle memory lots of us have built up over the years; having a little CLI like entry field we can use to quickly navigate to resource views might be quite handy; particularly understanding the CLI (resource/CRD) short names.

being inside octant means we don't have to type 

```bash 
kubectl get pod
```

we could just support inputs like:

```bash
po
po -n myns
po -l foo=bar
svc
pr -n cheese # for tekton PipelineRun
```

etc

**Comment by GuessWhoSamFoo** [comment]
That'd be a pretty cool idea. It also opens the door to having other tooling available like telepresence, kubectl plugins, etc almost out of the box.

Current implementation of terminals is coupled with a pod: https://github.com/vmware-tanzu/octant/issues/1751

Once that is refactored, it'd be fairly easy to build this out.

**Comment by jstrachan** [comment]
yeah; I'm not sure if it even needs to be an actual terminal per se. 

e.g. what if we had a "Search Box" icon on the left hand nav bar which if you click on it you get an input field.

![Octant@2x](https://user-images.githubusercontent.com/30140/112615689-b6acbd80-8e1a-11eb-9a1f-19853f69bd4b.png)

so the input field essentially lets you type the rest of the "kubectl get" command line (conceptually) then when you are ready you hit enter and it navigates to the right page to display those results?

it could easily understand the resource / CRD names and short names to do nice completion for you too with a little auto-complete box so that if you click the `Search Box` button and type `p`: you'd see something like

```bash 
kubectl get: p

persistentvolumeclaims     
persistentvolumes          
pipelineactivities         
pipelineresources          
pipelineruns               
pipelines                  
poddisruptionbudgets   
po    
pods                       
podsecuritypolicies        
podtemplates               
previews                   
pv
pvc
```
so typing a few characters would quickly help you select through any k8s resource/crd.

Once you've selected a resource kind maybe completion could help you either pick a name or label? (again could use completion like the CLI)


If we had this search box; maybe we could have a URL which encodes the search box text you enter (e.g. http://127.0.0.1:7777/#/myns/search?q=pods). If we had that then we could make a real CLI in the `octant` binary quite easily; so that typing

```bash
$> octant get pod cheese

opening: http://127.0.0.1:7777/#/myns/search?q=pod+cheese
```

so if you are really in a CLI then we'd have an `octant get` thats like `kubectl get` but which takes you to a page in octant that tries to view whatever resources you specify (with optional name / labels / filters / whatever)

**Comment by jstrachan** [comment]
even without the real CLI integration; having a `kubectl get` inspired quick search box to navigate to any k8s resource/CRD page would be super handy I think - lots of us use the CLI alot with k8s

**Comment by jstrachan** [comment]
@GuessWhoSamFoo an actual funky awesome terminal inside Octant is a cool idea too mind you ;)

**Comment by jstrachan** [comment]
DOH - I just noticed the quick switcher! (`Ctrl-Enter`) never noticed that before ;) - thats kinda the same thing I described above in many ways ;)

Though we could still do the CLI integration so you can open a page in Octant quickly from the CLI using a `kubectl get` like CLI for octant

**Comment by wwitzel3** [comment]
Related in many ways to #2247 

The idea being that Octant is able to create links that allow it to open specific resources - being able to provide context, namespace, filters and/or resource names and have that view open in Octant - be it from a CLI tool, or generating a link from the GUI - the desire is the same.
